bwt/wavelet.go

```diff
@@ -0,0 +1,340 @@
+package bwt
+
+import (
+	"fmt"
+	"math"
+
+	"golang.org/x/exp/slices"
+)
+
+// waveletTree datastructure that allows us to
+// conduct RSA queries on strings.
+//
+// For the waveletTree's usage, please read the its
+// method documentation. To understand what it is and how
+// it works, then read below.
+//
+// # The WaveletTree has several imporant components
+//
+// ## The Character's Path Encoding
+//
+// one important component is a character's path encoding.
+// which character we are woring with in a given path in the tree.
```

Comment:
typo: `woring` should be `working`

Reply:
Done. Also, I'll definitely be fixing all the others because of the lint.

With that being said, I'm not breaking Poly's bank when a push up these small commits am I :thinking: ? I can be more considerate if that is the case.